test/runtests.jl

```diff
@@ -1300,8 +1301,10 @@ end
     @test dot(Zeros(5), Ones{ComplexF16}(5)) ≡ zero(ComplexF64)
```

Comment:
```suggestion
    @test @inferred(dot(Zeros(5), Ones{ComplexF16}(5))) ≡ zero(ComplexF64)
```
So that we make sure the above is type-stable.

Reply:
also in the tests below